app/controllers/course/registrations_controller.rb

```diff
@@ -0,0 +1,158 @@
+class Course::RegistrationsController < Course::ModuleController
+  load_and_authorize_resource through: :course
+  before_action :ensure_course_open, :ensure_not_registered, only: :create
+
+  def index #:nodoc:
+    @staff_requests = @registrations.staff
+    @student_requests = @registrations.student
+
+    respond_to do |format|
+      format.html
+    end
+  end
+
+  def create #:nodoc:
+    role = (CourseUser.roles.include?(params[:role])) ? params[:role] : :student
+
+    existing = Course::Registration.find_by_user_id_and_course_id(current_user.id, @course.id)
+    if existing
+      existing
+      redirect_to update_course_registration_path(@course, existing,
+                                                  registration: existing.as_json,
+                                                  update_single: existing.id)
+      return
+    end
+    @registration.user = current_user
+    @registration.save!
+
+    # TODO: Notify lecturer
+    respond_to do |format|
+      format.html do
+        redirect_to course_path(@course), flash: { notice: t('.request_format') %
+          { role: t("course.registrations.role.#{role}") } }
+      end
+    end
+  end
+
+  def update #:nodoc:
+    if params[:update_single]
+      update_registration
+    elsif params[:approve_single]
+      approve_registration
+    elsif params[:delete_single]
+      redirect_to delete_course_registration_path(@course, @registration)
+    else
+      redirect_to course_path(@course), status: :bad_request
+    end
+  end
+
+  def delete #:nodoc:
+    message = t('.notice_format') % { role: t("course.registrations.role.#{@registration.role}"),
+                                      user: @registration.user.name }
+    @registration.destroy!
+
+    respond_to do |format|
+      format.html do
+        redirect_to course_registrations_path(@course), flash: { notice: message }
+      end
+    end
+  end
+
+  def update_multiple #:nodoc:
```

Comment:
Assignment Branch Condition size for update_multiple is too high. [25.36/15]<br>Method has too many lines. [19/10]